fix: consul watcher does not works on remote

```
2022-09-18T21:53:30.797Z [ERROR] watch.watch: Watch errored: type=key error="Get "http://127.0.0.1:8500/v1/kv/configuration/monitoring": dial tcp 127.0.0.1:8500: connect: connection refused" retry=20s
```
The following code does not work on remote consul. Change the following method to pass api in parameter and use it instead of default config:
```go
go func() {
		if err := plan.Run(api.DefaultConfig().Address); err != nil {
			log.Error(fmt.Errorf("err watching consul key: %v", err))
		}
	}()
```

--- a/Makefile
+++ b/Makefile
@@ -36,7 +36,7 @@ start:
 #test.unit: @ run unit tests and coverage
 test.unit:
 	@echo "[TEST.UNIT] run unit tests and coverage"
-	go test -race -covermode=atomic -coverprofile=coverage.out \
+	go test -covermode=atomic -coverprofile=coverage.out \
 		github.com/bigblueswarm/bigblueswarm/v2/pkg/admin \
 		github.com/bigblueswarm/bigblueswarm/v2/pkg/api \
 		github.com/bigblueswarm/bigblueswarm/v2/pkg/app \

--- a/pkg/config/config_test.go
+++ b/pkg/config/config_test.go
@@ -13,6 +13,12 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func stopWatcher() {
+	for i := 0; i < len(consulWatchers); i++ {
+		consulWatchers[i].Stop()
+	}
+}
+
 func TestFSConfigLoad(t *testing.T) {
 
 	type test struct {
@@ -75,6 +81,7 @@ func TestConsulConfigLoad(t *testing.T) {
 		}
 	}))
 
+	defer stopWatcher()
 	defer server.Close()
 
 	tests := []test.Test{

--- a/pkg/config/consul.go
+++ b/pkg/config/consul.go
@@ -12,6 +12,9 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+var consulConfig *api.Config
+var consulWatchers []*watch.Plan = []*watch.Plan{}
+
 // IsConsulEnabled check if path starts with ConsulPrefix (consul:)
 func IsConsulEnabled(path string) bool {
 	return strings.HasPrefix(path, ConsulPrefix)
@@ -28,8 +31,10 @@ func GetConsulConfig(path string) *api.Config {
 
 // LoadConfigFromConsul load BigBlueSwarm configuration from consul provider
 func LoadConfigFromConsul(path string) (*Config, error) {
+	consulConfig = GetConsulConfig(path)
+
 	// Get a new consul client
-	client, err := api.NewClient(GetConsulConfig(path))
+	client, err := api.NewClient(consulConfig)
 	if err != nil {
 		return nil, err
 	}
@@ -140,8 +145,10 @@ func WatchChanges(key string, handler func(value []byte)) error {
 		handler(pair.Value)
 	}
 
+	addWatcher(plan)
+
 	go func() {
-		if err := plan.Run(api.DefaultConfig().Address); err != nil {
+		if err := plan.Run(consulConfig.Address); err != nil {
 			log.Error(fmt.Errorf("err watching consul key: %v", err))
 		}
 	}()
@@ -252,4 +259,8 @@ func (c *Config) LoadInfluxDBConf(kv *api.KV) error {
 	}
 
 	return nil
+}
+
+func addWatcher(watcher *watch.Plan) {
+	consulWatchers = append(consulWatchers, watcher)
 }

